openapi-core/src/main/java/io/ballerina/openapi/core/generators/common/OASModifier.java

```diff
@@ -114,9 +115,110 @@ public OpenAPI modifyWithBallerinaConventions(OpenAPI openapi, Map<String, Strin
         openapi = modifyOASWithSchemaName(openapi, nameMap);
         modifyOASWithParameterName(openapi);
         modifyOASWithObjectPropertyName(openapi);
+        modifyOASWithObjectPropertyNameInlineObjectSchema(openapi);
         return openapi;
     }
 
+    private void modifyOASWithObjectPropertyNameInlineObjectSchema(OpenAPI openAPI) {
+        openAPI.getPaths().forEach((path, pathItem) -> processPathItem(pathItem));
+    }
+
+    private void processPathItem(PathItem pathItem) {
+        pathItem.readOperationsMap().forEach((method, operation) -> processOperationWithInlineObjectSchema(operation));
+    }
+
+    private void processOperationWithInlineObjectSchema(Operation operation) {
+        if (operation.getRequestBody() != null) {
+            updateInlineObjectInContent(operation.getRequestBody().getContent());
+        }
+        processParametersWithInlineObjectSchema(operation.getParameters());
+        processResponsesWithInlineObjectSchema(operation.getResponses());
+    }
+
+    private void processParametersWithInlineObjectSchema(List<Parameter> parameters) {
+        if (parameters == null) {
+            return;
+        }
+
+        for (Parameter parameter : parameters) {
+            if (parameter.getSchema() != null) {
+                updateInlineObjectSchema(parameter.getSchema());
+            }
+        }
+    }
+
+    private void processResponsesWithInlineObjectSchema(Map<String, ApiResponse> responses) {
+        responses.forEach((status, response) -> {
+            if (response.getContent() != null) {
+                updateInlineObjectInContent(response.getContent());
+            }
+        });
+    }
+
+    private void updateInlineObjectSchema(Schema<?> schema) {
+        if (schema == null) {
+            return;
+        }
+        handleInlineObjectSchemaInComposedSchema(schema);
+        handleInlineObjectSchema(schema);
+        handleInlineObjectSchemaItems(schema);
+        handleInlineObjectSchemaInAdditionalProperties(schema);
+        handleInlineObjectSchemaProperties(schema);
+    }
+
+    private void handleInlineObjectSchemaInComposedSchema(Schema<?> schema) {
+        if (schema instanceof ComposedSchema composedSchema) {
+            processSubSchemas(composedSchema.getAllOf());
+            processSubSchemas(composedSchema.getAnyOf());
+            processSubSchemas(composedSchema.getOneOf());
+        }
+    }
+
+    private void processSubSchemas(List<Schema> subSchemas) {
+        if (subSchemas != null) {
```

Comment:
```suggestion
        if (Objects.nonNull(subSchemas)) {
```
Change in other places as well